.github/actions/test-component/action.yml

```diff
@@ -35,6 +35,9 @@ runs:
     - name: Display Python version
       run: python -c "import sys; print(sys.version)"
       shell: bash
+    - name: Install OSX packages
```

Comment:
why are we changing the CI?

Reply:
this was an issue with xgboost where the system dependencies were not properly installed and compiler flags were not set. @yarnabrina suggested making a script that installs the required system packages and compiler flags.

Reply:
this impacts the entire CI - shoud we run a `test-all`, @yarnabrina?

Reply:
We can, but I think we'll be good with this. It's only adding system packages to cover xgboost tests in CI. Let's do this after your release CI is all done.